focuspoints.lrdevplugin/PentaxDelegates.lua

```diff
@@ -0,0 +1,180 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--]]
+
+--[[
+  A collection of delegate functions to be passed into the DefaultPointRenderer
+  for Pentax cameras.
+
+  Notes:
+  * Back focus button sets AFPointsInFocus to 'None' regardless of point
+    selection mode (sucks). AFPointSelected is set correctly with either button.
+
+  2017.03.29 - roguephysicist: works for Pentax K-50 with both phase and contrast detection
+--]]
+
+local LrStringUtils = import "LrStringUtils"
+local LrErrors = import 'LrErrors'
+require "Utils"
+
+-- LrErrors.throwUserError(result)
+
+
+PentaxDelegates = {}
+
+PentaxDelegates.focusPointsMap = nil
+PentaxDelegates.focusPointDimen = nil
+PentaxDelegates.metaKeyFocusMode = {"Focus Mode"}
+
+function PentaxDelegates.getAfPoints(photo, metaData)
+  local focusMode = ExifUtils.findFirstMatchingValue(metaData, PentaxDelegates.metaKeyFocusMode)
+  if (string.find(focusMode, 'AF%-S') ~= nil) then
+    results = PentaxDelegates.getAfPointsPhase(photo, metaData)
+  elseif (string.find(focusMode, 'Contrast%-detect') ~= nil) then
+    results = PentaxDelegates.getAfPointsContrast(photo, metaData)
+  end
+  return results
+end
+
+--[[
+-- photo - the photo LR object
+-- metaData - the metadata as read by exiftool
+--]]
+function PentaxDelegates.getAfPointsPhase(photo, metaData)
+  local afPointsSelected = ExifUtils.findFirstMatchingValue(metaData, { "AF Point Selected" })
+  if afPointsSelected == nil then
+    afPointsSelected = {}
+  else
+    afPointsSelected = splitTrim(afPointsSelected, ";") -- pentax separates with ';'
+  end
+  local afPointsInFocus = ExifUtils.findFirstMatchingValue(metaData, { "AF Points In Focus" })
+  if afPointsInFocus == nil then
+    afPointsInFocus = {}
+  else
+    afPointsInFocus = splitTrim(afPointsInFocus, ",")
+    afPointsInFocus = PentaxDelegates.fixCenter(afPointsInFocus)
+  end
+    
+  local result = {
+    pointTemplates = DefaultDelegates.pointTemplates,
+    points = {}
+  }
+
+  for key,value in pairs(PentaxDelegates.focusPointsMap) do
+    local x = PentaxDelegates.focusPointsMap[key][1]
+    local y = PentaxDelegates.focusPointsMap[key][2]
+    local width = PentaxDelegates.focusPointDimen[1]
+    local height = PentaxDelegates.focusPointDimen[2]
+    
+    local pointType = DefaultDelegates.POINTTYPE_AF_INACTIVE
+    local isInFocus = arrayKeyOf(afPointsInFocus, key) ~= nil
+    local isSelected = arrayKeyOf(afPointsSelected, key) ~= nil
+    if isInFocus and isSelected then
+      pointType = DefaultDelegates.POINTTYPE_AF_SELECTED_INFOCUS
+    elseif isInFocus then
+      pointType = DefaultDelegates.POINTTYPE_AF_INFOCUS
+    elseif isSelected then
+      pointType = DefaultDelegates.POINTTYPE_AF_SELECTED
+    end
+
+    if width > 0 and height > 0 then
```

Comment:
I would think LR exits with an error if width and height can't be found. Can you try it and verify? I wouldn't think that <=0 would be returned from a array that doesn't have a key.